pool/hub.go

```diff
@@ -109,6 +126,21 @@ type Hub struct {
 	wg             *sync.WaitGroup
 }
 
+// SetNodeConnection sets the mining node connection.
```

Comment:
Could the client connections be passed as parameters to `pool.NewHub()`?

That would allow us to remove the setter functions, and also we could check `if nodeConn==nil` just once in `newHub()` - we wouldn't have to duplicate the check in `getBlock()`, `getWork()` and `submitWork()`.

Reply:
The notification handlers require an initialized hub to create in `CreateNotificationHandlers()`.